### PR TITLE
Signin

### PR DESCRIPTION
**Pull request by YesicaNavaB**
## Summary

Sign In

## Trello Card

https://trello.com/c/2N2BxfoK

### CODE REVIEW

File: app/errors.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ exports.albumsApiError = message => internalError(message, exports.ALBUMS_API_ER
 exports.SIGN_UP_ERROR = 'sign_up_error';
 exports.signUpError = message => internalError(message, exports.SIGN_UP_ERROR);
 
+exports.SIGN_IN_ERROR = 'sign_in_error';
+exports.signInError = message => internalError(message, exports.SIGN_IN_ERROR);
+
 exports.VALIDATE_USER = 'validate_user';
 exports.validateUser = message => internalError(message, exports.VALIDATE_USER);
```
Comment (jersonlopez) [inline app/errors.js]: `validateUserError` for the sintaxis please!

File: app/middlewares/errors.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
   [errors.ALBUMS_API_ERROR]: 503,
   [errors.SIGN_UP_ERROR]: 400,
+  [errors.SIGN_IN_ERROR]: 400,
```
Comment (jersonlopez) [inline app/middlewares/errors.js]: I think the code is 401, what do you think?

File: app/middlewares/users.js [hunk 3]
```diff
@@ -37,3 +37,21 @@ exports.signUpMiddleware = [
     .withMessage('Must be at least 8 chars long'),
   validateRequest
 ];
```
Comment (jersonlopez) [inline app/middlewares/users.js:39]: insert \n, is more pretty

File: app/services/users.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 const { User } = require('../models'),
   error = require('../errors'),
-  logger = require('../logger');
+  logger = require('../logger'),
+  bcrypt = require('bcryptjs'),
+  jwt = require('jwt-simple'),
+  secret = 'xxx';
```
Comment (jersonlopez) [inline app/services/users.js]: `secret` must a environment variable

File: app/services/users.js [hunk 5]
```diff
@@ -16,3 +19,17 @@ exports.createUser = userData =>
       logger.error(`Could not create user: ${userData.name}`);
       throw error.databaseError(err.message);
     });
+
+exports.signIn = data =>
```
Comment (jersonlopez) [inline app/services/users.js]: apply destructuting here ({ email, password }) => ...

File: app/middlewares/users.js [hunk 6]
```diff
@@ -37,3 +37,21 @@ exports.signUpMiddleware = [
...
+    .withMessage('email is required')
+    .isEmail()
+    .matches(/^(([^<>\\[\]\\.,;:\s@"]+(\.[^<>()\\[\]\\.,;:\s@"]+)*)|(".+"))@wolox.(co|com|com.ar)\s*$/)
+    .withMessage('email is not valid or does not belong to the wolox domain'),
+  check('password')
+    .not()
+    .isEmpty()
+    .withMessage('password is required')
+    .isAlphanumeric()
+    .withMessage('Must be only alphanumeric chars')
```
Comment (jersonlopez) [inline app/middlewares/users.js]: this message is unnecessary for the login

File: app/middlewares/users.js [hunk 7]
```diff
@@ -37,3 +37,21 @@ exports.signUpMiddleware = [
...
+    .isLength({ min: 8 })
+    .withMessage('Must be at least 8 chars long'),
```
Comment (jersonlopez) [inline app/middlewares/users.js]: and this too.

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: test/users.test.js [hunk 8]
```diff
@@ -82,3 +84,43 @@ describe('User registration test, with their respective fields', () => {
...
+        .send({ email: 'yesica@wolox.co', password: 'shdfgs345' })
+        .set('Accept', 'application/json')
+        .then(response => {
+          const token = {
+            token:
+              'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6Inllc2ljYUB3b2' +
+              'xveC5jbyJ9.W94vf6ymuks9qEsz-dDciig304QtAa7FeUjlNqwXaI8'
+          };
+          expect(response.statusCode).toBe(200);
+          console.log(response.text);
```
Comment (jersonlopez) [inline test/users.test.js]: what is doing this here? jaajaajaj 😂😂😂😂😂😂

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: test/users.test.js (hunk 8 above)
Comment (YesicaNavaB) [reply test/users.test.js]: Sorry! 😂

**Review by jersonlopez — APPROVED** [review]
good job 🥇

**Pull request merged**